Add GH action for release PR & checklist

tomalec opened the pull request:
_This is a continuation of https://github.com/woocommerce/automatewoo/pull/1404_
### Changes proposed in this Pull Request:

Adds a new GitHub Action to prepare extension release.

To be used with a manual GH workflow. It takes versions, then starts a branch and creates a PR with the checklist.
The aim is to automate and unify the release process across our repos.


### Screenshots:

![image](https://user-images.githubusercontent.com/17435/233470959-7affcc0a-201d-49c7-9172-ae993bb055ea.png)


### Detailed test instructions:
<!-- Add detailed instructions for how to test that this PR fixes the issue and confirm that it doesn't break any other features :) -->

1. Add a workflow in another repo following the instructions from `packages/js/github-actions/actions/prepare-extension-release/README.md`, replacing `woocommerce/grow/prepare-extension-release@actions-v1` with `woocommerce/grow/prepare-extension-release@actions-v1.5.2-pre` as this PR is not yet released.

   Like https://github.com/tomalec/grow/blob/391113cd2a00bac2dc808674ecebdb30e8fd83da/.github/workflows/extension-release.yml
2. Run the created workflow. 

### Additional details:

The further improvements automation I was trying to implement but didn't finish:
1. (tricky) Make the workflow run `woorelease simulate`. So we could reduce two more manual steps for the developer, and serve them just a `.zip` package to smoke test & changelog to inspect. It's tricky, as requires setting up all woorelease keys on GHA, and makes the simulation repo harder to inspect.
2. (tricky) Make another workflow pick up the approval, and make the actual release. The tricky part is as above. But it may mitigate a few more issues we were experiencing in the past, as it could assert the consistent build environment (node, npm, composer versions)
3. (easier) Make another workflow pick up a release, then 
   1. publish the GitHub's release notes as a comment to matching PR,
   2. merge the PR to `trunk`
   3. merge `trunk` to `develop`

Further improvements I envision:
1. Allow bumping version by just `major|minor|patch` instead of typing a full version
2. Allow customizing checklist, so some repos could have a few more specific steps


<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with change type prefix`(Fix|Add|…) - `, for example:
> Break - A change breaking previous API or functionality.
> Add - A new feature, function or functionality was added.
> Update - Big changes to something that wasn't broken.
> Fix - Took care of something that wasn't working.
> Tweak - Small change, that isn't actually very important.
> Dev - Developer-facing only change.
> Doc - Updated customer or developer facing documentation

If you remove the "Changelog entry" header, the title of Pull Request will be used as the changelog entry.

Add the `changelog: none` label if no changelog entry is needed.
-->
### Changelog entry

> Add - Extension release preparation github-action.

tomalec committed:
Add GH action for release PR & checklist

tomalec committed:
Add `prepare-extension-release` to github-actions/README.md

tomalec committed:
Remove not used section from prepare-ext…/README

mikkamp commented on packages/js/github-actions/actions/prepare-extension-release/src/woo-extension-create-pr-for-release.mjs:
```diff
@@ -0,0 +1,60 @@
+   1. [ ] Run a few basic smoke tests
+
+## Next steps
+1. ~Approve this PR to allow the next workflow to create an actual release.~
```
Just curious, which workflow is this pointing to? Or is this still something we need to create?

tomalec replied:
Something to be implemented, see https://github.com/woocommerce/grow/pull/56#discussion_r1210713706

mikkamp commented on packages/js/github-actions/actions/prepare-extension-release/src/woo-extension-create-pr-for-release.mjs:
```diff
@@ -0,0 +1,60 @@
+   When prompted for changelog entries, double-check and apply any changes if needed.
+1. ~Another workflow will pick up the new release and post a comment here with GitHub's release notes, merge the PR, and merge \`trunk\` to \`develop\`.~
+1. [ ] Go to ${ context.payload.repository.html_url }/releases/${ version }, generate GitHub release notes, and paste them as a comment here.
+1. [ ] Merge this PR after the new release is successfully created and the version tags are updated.
```
In two steps above this one we mention that a workflow will run to merge the PR and merge trunk to develop. Yet here it mentions to merge the PR but doesn't mention to merge trunk to develop. From what I understand the workflow hasn't been created yet, but until then can we mention all the manual steps that are needed. Or clarify what needs to be done here.

tomalec replied:
I created strike-through items to keep an eye on workflows to be implemented, mostly because I was planning to do at least some of them within HACK day. But I'll remove them as it seems they introduce more confusion than value :)

mikkamp reviewed (approved):
Looks good to me. I think this will continue to be a process to iterate on, but we aren't going to do that until we merge it and start using it.

I left a few comments but they aren't blockers as we should start the adoption and improve as we go along.

tomalec committed:
Remove striked next steps from `prepare-extension-release`,

to reduce confusion. These were to be implemented ideas.

Add manual next stepto merge `trunk` to `develop`.

Addresses https://github.com/woocommerce/grow/pull/56#discussion_r1210209902
